### PR TITLE
feature/CasysAndPreprocessingAndXAlign2

### PR DESCRIPTION
**Pull request by Sandy-CHERY**
## Description

**Fixes 1 Change Cassys to solve the max stack size reached problem**

## Motivation and Context

Alexis Neme have a problem :
"When I execute Cassys  command (below) with IDE it generates the error "Max stack size reached" (attcahed file)
"Local\Unitex\App\UnitexToolLogger.exe" Cassys "-aUnitex3_2\English\Alphabet.txt" "-tUnitex3_2\English\Corpus\text.snt" "-lUnitex3_2\English\CasSys\cascade.csc" -v "-rUnitex3_2\English\Graphs\" "--input_offsets=Unitex3_2\English\Corpus\text_snt\normalize.out.offsets" -qutf8-no-bom

When I execute Cassys  command (below) with a command line, it does generate an error
"Local\Unitex\App\UnitexToolLogger.exe" Cassys "-aUnitex3_2\English\Alphabet.txt" "-tUnitex3_2\English\Corpus\text.snt" "-lUnitex3_2\English\CasSys\cascade.csc" -v "-rUnitex3_2\English\Graphs\" "--input_offsets=Unitex3_2\English\Corpus\text_snt\normalize.out.offsets" -qutf8-no-bom --locate_argument=--stack_max=8000 --locate_argument=--max_matches_per_subgraph=800 --locate_argument=--max_matches_at_token_pos=8000

I propose to include these parameters in bold to the Cassys command line call from the IDE"

## What I did

In gramlab-ide\unitex\src\fr\umlv\unitex\process\commands\CommandBuilder.java,
when we create the Cassys comand "-qutf8-no-bom", I add "-qutf8-no-bom --locate_argument=--stack_max=8000 --locate_argument=--max_matches_per_subgraph=800 --locate_argument=--max_matches_at_token_pos=8000"

## Description

**Fixes 2 Preprocessing**

## Motivation and Context

#105 opened on 28 May 2018 by denisMaurel : SET (line Merge) in Preprocessing window type:feature-request

When you lanch the Preprocessing, graph in MERGE mode and graph in REPLACE mode have the same path (Graphs\Preprocessing\Replace)

## What I did

In gramlab-ide\unitex\src\fr\umlv\unitex\frames\PreprocessDialog.java
Sentence and Replace used the same variable, I creat two different variables for having two different paths (Graphs\Preprocessing\Sentence and Graphs\Preprocessing\Replace).

## Description

**Fixes 3 Change XAlign problem**

## Motivation and Context

When you lanch XAlign, you need to choose two text, Source and Target. If texts are in .txt, you need to choose the .XML text. but they have the same path.

## What I did

In gramlab-ide\unitex\src\fr\umlv\unitex\frames\XAlignConfigFrame.java
Source and Target used the same variable saveXMLChooser, I creat two different variables for having two different paths.

### CODE REVIEW

**Comment by Sandy-CHERY** [comment]
I think I did it.
Can you please check ?

File: unitex/src/fr/umlv/unitex/process/commands/AbstractCommand.java [hunk 1]
```diff
@@ -1,50 +1,50 @@
-/*
```
Comment (martinec) [inline unitex/src/fr/umlv/unitex/process/commands/AbstractCommand.java:1]: In all the files, either the space characters (spaces->tabs) or the newline characters (lf->crlf) are being changed. So it is not clear which changes you want to merge.

**Review by martinec — CHANGES_REQUESTED** [review]
As a general rule, you should not refactor spaces/new lines and always check that your code compiles before submitting a PR.

**Comment by Sandy-CHERY** [comment]
Bonjour,
Tu parles bien d'un test sur le lancement de Cassys ?
Ici (sous Windows) je ne note pas de différences entre la version stable et celle que j'ai modifiée. La modification consiste juste à augmenter la mémoire du 'locate' lancé par Cassys :
dans ComandBuilder, j'ajoute "--locate_argument=--stack_max=8000 --locate_argument=--max_matches_per_subgraph=800 --locate_argument=--max_matches_at_token_pos=8000\r\n" à la requête appelant Cassys.

**Comment by martinec** [comment]
> Bonjour,
> Tu parles bien d'un test sur le lancement de Cassys ?
> Ici (sous Windows) je ne note pas de différences entre la version stable et celle que j'ai modifiée. La modification consiste juste à augmenter la mémoire du 'locate' lancé par Cassys :
> dans ComandBuilder, j'ajoute "--locate_argument=--stack_max=8000 --locate_argument=--max_matches_per_subgraph=800 --locate_argument=--max_matches_at_token_pos=8000\r\n" à la requête appelant Cassys.

The PR is trying to modified more than 6000 !!! lines of code.

For example [here](https://github.com/UnitexGramLab/gramlab-ide/pull/156/files#diff-88ca56b3217775d9dc2e95b39e9df54f419ad2309cfbaf84edfff581625b87dfL2), you are removing the following code: 

deletions

To add the following:

additions

As both codes are exactly the same, i'm supposing that either the space characters (spaces->tabs) or the newline characters (lf->crlf) are being changed. So it is not clear which changes you want to merge.

**Pull request closed without merging**